lldb/source/Target/StackID.cpp

```diff
@@ -61,6 +61,11 @@ bool lldb_private::operator<(const StackID &lhs, const StackID &rhs) {
   const lldb::addr_t lhs_cfa = lhs.GetCallFrameAddress();
   const lldb::addr_t rhs_cfa = rhs.GetCallFrameAddress();
 
+  // Check for async -> sync comparison. (stack: high, heap: low)
+  if (lhs_cfa - rhs_cfa >= 0x00007ff000000000ULL) {
```

Comment:
This encodes ABI knowledge in a non-ABI-specific class. If there is no better general solution, this should call into a ABI-specific class here. Hopefully there's something better we can do, perhaps asking the Unwinder.

Reply:
I have no plans to commit this as is. With more thinking and people's input we'll find a better way.

Reply:
Is it possible for code in an async stack could make another async call?  In that case both bits of the stack would be in the heap and I don't think you would be able to use addresses to determine the ordering at all.